File: demo/components/tooltip/tooltip-demo.ts
import { ChangeDetectionStrategy, Component } from '@angular/core';

// webpack html imports
let template = require('./tooltip-demo.html');

@Component({
  selector: 'tooltip-demo',
  template: template,
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class TooltipDemoComponent {
  public dynamicTooltip:string = 'Hello, World!';
  public dynamicTooltipText:string = 'dynamic';
  public htmlTooltip:string = 'I\'ve been made <b>bold</b>!';
  public tooltipModel:any = {text: 'foo', index: 1};

  public tooltipStateChanged(state: boolean):void {
    console.log(`Tooltip is open: ${state}`);
  }
}

File: web/components/tooltip/tooltip-demo.dart
import "package:angular2/angular2.dart";
import 'package:ng2_strap/tooltip/tooltip.dart';

@Component(selector: "tooltip-demo")
@View(
    templateUrl: 'tooltip-demo.html',
    directives: const [TOOLTIP_DIRECTIVES, CORE_DIRECTIVES, FORM_DIRECTIVES, NgClass],
    styles: const [
      '''
    /* Specify styling for tooltip contents */
    .tooltip.customClass .tooltip-inner {
        color: #880000;
        background-color: #ffff66;
        box-shadow: 0 6px 12px rgba(0,0,0,.175);
    }
    /* Hide arrow */
    .tooltip.customClass .tooltip-arrow {
        display: none;
    }
  '''
    ])
class TooltipDemo {
  String dynamicTooltip = "Hello, World!";
  String dynamicTooltipText = "dynamic";
  String htmlTooltip = "I've been made <b>bold</b>!";
  String inputModel;
}
